project/Dependencies.scala

```diff
@@ -71,6 +74,24 @@ object Dependencies {
     }
   }
 
+  def json4sVersionForScala(scalaVersion: String): String = {
+    scalaVersion match {
+      case _ if scalaVersion.startsWith("2.11") => Versions.json4s_spark2
+      case _ if scalaVersion.startsWith("2.12") => Versions.json4s_spark3
+      case _ => throw new IllegalArgumentException("Only Scala 2.11 and 2.12 are currently supported.")
+    }
+  }
+
+  def moduleByScala(moduleIdWithoutVersion: String => ModuleID)
+                   (scala211Version: String, scala212Version: String)
+                   (actualScalaVersion: String): ModuleID = {
+    actualScalaVersion match {
+      case _ if actualScalaVersion.startsWith("2.11") => moduleIdWithoutVersion.apply(scala211Version)
+      case _ if actualScalaVersion.startsWith("2.12") => moduleIdWithoutVersion.apply(scala212Version)
```

Comment:
This is becoming a common occurrence in the project, maybe a dedicated function returning Scala major + Scala minor, that could be directly matched would come handy.

Reply:
This function is actually not even used. I will remove it.